src/server/DeviceServer_v2.js

```diff
@@ -263,6 +267,54 @@ class DeviceServer {
     await this._eventPublisher.publish(eventData);
   };
 
+  _onDeviceSubscribe = async (
+    message: Message,
+    device: SparkCore,
+  ): Promise<void> => {
+    const deviceID = device.getHexCoreID();
+    // uri -> /e/?u    --> firehose for all my devices
+    // uri -> /e/ (deviceid in body)   --> allowed
+    // uri -> /e/    --> not allowed (no global firehose for cores, kthxplox)
+    // uri -> /e/event_name?u    --> all my devices
+    // uri -> /e/event_name?u (deviceid)    --> deviceid?
+    const messageName = message.getUriPath().substr(3);
+
+    if (!messageName) {
+      device.sendReply('SubscribeFail', message.getId());
+      return;
+    }
+
+    const query = message.getUriQuery();
+    const fromMyDevicesOnly = query && !!query.match('u');
```

Comment:
not sure about name of this var. what is this for. it checks if device wants to listen event from devices with the same ownerID as it has. `Particle.subscribe( lall, lulu, MY_DEVICES) ` - in firmware